windows-agent/internal/config/config.go

```diff
@@ -300,99 +243,56 @@ type RegistryData struct {
 func (c *Config) UpdateRegistryData(ctx context.Context, data RegistryData, db *database.DistroDB) (err error) {
 	defer decorate.OnError(&err, "config: could not update registry-provided data")
 
-	taskList, err := c.collectRegistrySettingsTasks(ctx, data, db)
-	if err != nil {
-		return err
-	}
-
-	// Apply tasks for updated settings
-	for _, d := range db.GetAll() {
-		err = errors.Join(err, d.SubmitDeferredTasks(taskList...))
-	}
-
-	if err != nil {
-		return fmt.Errorf("could not submit new task to certain distros: %v", err)
-	}
-
-	return nil
-}
-
-// collectRegistrySettingsTasks looks at the registry data to see if any of them have changed since this
-// function was last called. It returns a list of tasks to run triggered by these changes, and updates
-// the config.
-func (c *Config) collectRegistrySettingsTasks(ctx context.Context, data RegistryData, db *database.DistroDB) (taskList []task.Task, err error) {
-	type getTask = func(*Config, context.Context, RegistryData, *database.DistroDB) (task.Task, error)
-
-	defer decorate.OnError(&err, "could not generate tasks from updated registry settings")
+	// We must perform the notification outside the lock to avoid deadlocks
+	afterUnlock := []func(){}
```

Comment:
For a moment I thought I was reading a C++ lambda 🤭

Reply:
heretic! :p